src/services/blocks/BlocksService.ts

```diff
@@ -450,6 +469,123 @@ export class BlocksService extends AbstractService {
 		}
 	}
 
+	private async getPartialFeeInfo(
+		events: ISanitizedEvent[],
+		extrinsicHex: string,
+		hash: BlockHash,
+		getFeeByEvent: boolean
+	) {
+		const { api } = this;
+		const { class: dispatchClass, partialFee } =
+			await api.rpc.payment.queryInfo(extrinsicHex, hash);
+
+		/**
+		 * Check if we should retrieve the partial_fee from the Events
+		 */
+		let fee: Balance | string = partialFee;
+		let error: string | undefined;
+		if (getFeeByEvent) {
+			// Event::Withdraw
+			const feeInfo = this.getPartialFeeByEvents(events, partialFee);
+			fee = feeInfo.partialFee;
+			error = feeInfo.error;
+		}
+
+		return {
+			partialFee: fee,
+			dispatchClass,
+			error,
+		};
+	}
+
+	private getPartialFeeByEvents(
+		events: ISanitizedEvent[],
+		partialFee: Balance
+	): { partialFee: string; error?: string } {
+		// Check Event:Withdraw event for the balances pallet
+		const withdrawEvent = events.find(
+			({ method }) =>
+				isFrameMethod(method) &&
+				method.method === Event.withdraw &&
+				method.pallet === 'balances'
+		);
+		if (withdrawEvent && withdrawEvent.data) {
+			const eventArr = withdrawEvent.data.toJSON();
+			if (Array.isArray(eventArr)) {
+				const fee = (eventArr as Array<number>)[eventArr.length - 1];
+				const diff = subIntegers(new BN(fee), partialFee).toString();
+
+				// The difference between values is 00.00001% or less so they are alike.
+				if (
```

Comment:
It's this comparison right here that I would like some input on. What is a better way to ensure that the given value is indeed an inclusion. I do think comparing the estimated value from `rpc::payment::queryInfo` is a good approach but mathematically what constraints do we think are viable or accurate enough?

@joepetrowski @IkerAlus @jsdw

Reply:
Important to note here too, this is not regarding parity based chains such as polkadot, kusama, and westend. But external chains and parachains.

Reply:
Aah, I think I'd assumed that `feeByEvent` would be a query param that accepted something like `?feeByEvent=balances.withdraw.4` to point at the event/property-in-event to pull a fee out of (for when parachains know what events they want to be inspected to find the actual fee).

But I guess the point here is that when a fee is taken, we probably expect one of these specific events to signal that some token has been taken for the fee? But this might vary based on the parachain and how they implement these things?

Code wise, it feels a bit "messy" to have to guess like this; we're essentialy saying "hmm this number looks about right so we'll use that!". But I appreciate that there may be no better way until that more standard TransactionFee event is a thing? What about making them point at the specific event they want us to use?

We could cut down on some repetition too (a `function findEvent(events, pallet, method)` and a `function areNumbersCloseEnough(a, b)` sort of thing for instance) :)

Reply:
> Aah, I think I'd assumed that feeByEvent would be a query param that accepted something like ?feeByEvent=balances.withdraw.4 to point at the event/property-in-event to pull a fee out of (for when parachains know what events they want to be inspected to find the actual fee).

Per our conversation offline, it makes sense to keep the current implementation. As the guesswork is not perfectly ideal, it will allow for support for historic blocks. Once `TransactionPaidEvent` is in, then we can set that as the default event to retrieve fee's from and keep this implementation as a fallback for historic blocks as mentioned above. 

> We could cut down on some repetition too (a function findEvent(events, pallet, method) and a function areNumbersCloseEnough(a, b) sort of thing for instance) :)

Yea I will try to clean it up but each event has the `Balance` value potentially in a different location in the data array, but there might be a good way to organize it more.

Reply:
@jsdw So I actually took your suggestions and cleaned up quite a bit. I think it's way better now. As soon as this PR goes in, I will add the `TransactionPaidFee` event as well.